test/services/SubmissionServiceSpec.scala

```diff
@@ -41,156 +43,286 @@ class SubmissionServiceSpec
+        s"register Without Business Verification" in {
+
+          val journeyConfigWithoutBV = testSoleTraderJourneyConfig.copy(businessVerificationCheck = false)
+
+          mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetails)))
+          mockMatchSoleTraderDetails(testJourneyId, testIndividualDetails, journeyConfigWithoutBV)(Future.successful(Right(true)))
+          mockStoreIdentifiersMatch(testJourneyId, identifiersMatch = true)(Future.successful(SuccessfullyStored))
+          mockRegisterWithoutBusinessVerification(testJourneyId, testIndividualDetails.optNino, testIndividualDetails.optSautr.get)(Future.successful(Registered("someId")))
+
+          val result = await(TestService.submit(testJourneyId, journeyConfigWithoutBV))
+
+          result mustBe JourneyCompleted(journeyConfigWithoutBV.continueUrl + s"?journeyId=$testJourneyId")
+
+        }
+      }
+
+    "the user has no sautr" should {
+      s"not be registered"  in {
+        val matchingResult = Random.nextBoolean()
+
+        val journeyConfigWithoutBV = testSoleTraderJourneyConfig.copy(businessVerificationCheck = false)
+
+        mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetailsNoSautr)))
+        mockMatchSoleTraderDetails(testJourneyId, testIndividualDetailsNoSautr, journeyConfigWithoutBV)(Future.successful(Right(matchingResult)))
+        mockStoreIdentifiersMatch(testJourneyId, identifiersMatch = true)(Future.successful(SuccessfullyStored))
+        mockStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)(Future.successful(SuccessfullyStored))
+
+        val result = await(TestService.submit(testJourneyId, journeyConfigWithoutBV))
+
+        result mustBe JourneyCompleted(journeyConfigWithoutBV.continueUrl)
+
+        verifyStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)
+
+      }
+    }
+
+    "the user has no Nino and enable(EnableNoNinoJourney)" should {
+      s"not register but create a trn"  in  {
+        val matchingResult = Random.nextBoolean()
+
+        enable(EnableNoNinoJourney)
+        val journeyConfigWithoutBV = testSoleTraderJourneyConfig.copy(businessVerificationCheck = false)
+
+        mockRetrieveIndividualDetails(testJourneyId)(Future.successful(Some(testIndividualDetailsNoNinoNoSautr)))
+        mockMatchSoleTraderDetailsNoNino(testJourneyId, testIndividualDetailsNoNinoNoSautr)(Future.successful(Right(matchingResult)))
+        mockCreateTrn(testJourneyId)(Future.successful(testTrn))
+        mockStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)(Future.successful(SuccessfullyStored))
+
+        val result = await(TestService.submit(testJourneyId, journeyConfigWithoutBV))
+
+        result mustBe JourneyCompleted(journeyConfigWithoutBV.continueUrl)
+
+        verifyStoreRegistrationResponse(testJourneyId, RegistrationNotCalled)
+
+      }
+    }
+
+    "the user has no Nino and disable(EnableNoNinoJourney)" should {
```

Comment:
This should never happen. To skip the nino question the Feature Switch has to be enabled so it would be very unexpected if the user got through the flow!

Reply:
here we need to talk ...... when you write:
 if (individualDetails.optNino.isEmpty && isEnabled(EnableNoNinoJourney)) .... do you mean:
EnableNoNinoJourney and therefore you expect optNino empty????
this mean that if you EnableNoNinoJourney and optNino.isDefined it will be an error and we have to throw an exception ......  we have to talk for this.... basically I wrote a test saying ......

"for individual journey: EnableNoNinoJourney and a Nino defined" should {
      "be not possible and throws an exception" in { ..... }

Reply:
A nino can be defined no matter the state of the feature switch. But it can only be missing from the user's data if the EnableNoNinoJourney Feature switch is enabled.

Reply:
Now I understood ... I think :(

Reply:
I rewrite it ... let's see after lunch if you agree with it